proposals/0339-increase-cpi-account-info-limit.md

```diff
@@ -0,0 +1,64 @@
+---
+simd: '0339'
+title: Increase CPI Account Info Limit
+authors:
+  - Justin Starry (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2025-08-15
+feature: (fill in with feature key and github tracking issues once accepted)
+---
+
+## Summary
+
+Increase the maximum account info length for cross-program invoked (CPI)
+instructions from 64 to 255 so that onchain programs can invoke CPI's without
+first needing to deduplicate account info's.
+
+## Motivation
+
+CPI's are restricted to a limit of 64 account info's passed to the syscall.
```

Comment:
Didn't you already raise the limit in https://github.com/solana-labs/solana/pull/27242?

The feature gate is still there: https://github.com/anza-xyz/agave/blob/d54feac498c6ce29fce9b4b262ea3ba1d3bb5a5e/syscalls/src/cpi.rs#L932

Reply:
Yes, thanks for pointing it out. The `increase_tx_account_lock_limit` feature gate is currently only activated on devnet and it does two things:
1. Increase tx account lock limit from 64 to 128
2. Increase CPI account info limit from 64 to 128

SIMD-0339 should change the CPI account info limit to 255 no matter whether the `increase_tx_account_lock_limit` feature gate is active or not. This SIMD has nothing to do with the tx account lock limit. It's more of an acknowledgement that account info length can contain duplicates and should be allowed to be as long as the ix account length and not necessarily the same as the tx account lock limit.